server/src/model.ts

```diff
@@ -907,6 +931,89 @@ async function createDocumentVersion(docId: Uint8Array): Promise<{
   return docVersion;
 }
 
+function filterViewableActivity(loggedInUserId: Uint8Array, isAdmin: boolean) {
+  // For an activity to be viewable, one of these conditions must be true:
+  // 1. You are the owner
+  // 2. The activity is public
+  // 3. The activity is shared with you
+  // 4. You are an admin and the activity is in the library
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const visibilityOptions: any[] = [
+    { ownerId: loggedInUserId },
+    { isPublic: true },
+    { sharedWith: { some: { userId: loggedInUserId } } },
+  ];
+  if (isAdmin) {
+    visibilityOptions.push({ owner: { isLibrary: true } });
+  }
+  return {
+    isDeleted: false,
+    isFolder: false,
+    OR: visibilityOptions,
+  };
+}
+
+function filterEditableActivity(loggedInUserId: Uint8Array, isAdmin: boolean) {
+  return {
+    ...filterEditableContent(loggedInUserId, isAdmin),
+    isFolder: false,
+  };
+}
+
+function filterEditableContent(loggedInUserId: Uint8Array, isAdmin: boolean) {
```

Comment:
We should up our game and start adding doc strings with JSDoc conventions `/** ... */`  on our functions. The resulting pop-up descriptions while editing can be helpful.